src/functions/merge.js

```diff
@@ -2,11 +2,45 @@
 
 function clone(value) {
   if (typeof value === 'object' && value !== null) {
-    return merge(Array.isArray(value) ? [] : {}, value);
+    return _merge(Array.isArray(value) ? [] : {}, value);
   }
   return value;
 }
 
+function isObjectOrArray(value) {
+  return (
+    typeof value === 'function' ||
```

Comment:
I think it would be clearer for the reader to reflect in the function's name that we also return `true` for functions.
`isObjectOrArrayOrFunction` ?

Reply:
done here: 
87c9024